runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -480,6 +505,21 @@ public synchronized StreamManager getStreamManager() {
         return this.streamManager;
     }
 
+    /**
+     * Register a streaming subscription manager as a singleton.
+     */
+    public StreamingManager getStreamingManager() {
```

Comment:
You can improve the read performance by introducing a local variable. For example, 

```java
public StreamingManager getStreamingManager() {
        StreamingManager localRef = streamingManager;
        if (localRef == null) {
            synchronized (this) {
                localRef = streamingManager;
                if (localRef == null) {
                    localRef = streamingManager = new StreamingManager(runtime);
                }
            }
        }
        return localRef;
    }
```

Reply:
Alternately, you can just use Lomboks lazy annotation and it would emit the code for double checked locking.

```java
    @Getter(lazy = true)
    private final StreamingManager streamingManager = create();
    private StreamingManager create() {
        return new StreamingManager(runtime);
    }
```

Reply:
Since this is not the hot path, I switched to use the previous way that leverages synchronizes the method itself, to make it simpler as we have another shutdown() method that needs to check if it's null or not.